### DISCUSSION TITLE
20220901 developer call notes

### DISCUSSION THREAD

**Discussion by hammer** [sgkit-dev/sgkit#900] [Meeting Notes]
### Pre-notes

* [Notes](https://github.com/pystatgen/sgkit/discussions/896) from previous meeting
* R: What’s the plan for the performance analysis - is this something I could help with?

#### PRs

* [sgkit PR#891](https://github.com/pystatgen/sgkit/pull/891) (Tim) Pedigree inbreeding
* [sgkit PR#892](https://github.com/pystatgen/sgkit/pull/892) (Tsz-Fung Chan) Add mean imputation function
* [sgkit PR#895](https://github.com/pystatgen/sgkit/pull/895) (Tim) Replace additive_relationships with pedigree_relationship
* [sgkit PR#899](https://github.com/pystatgen/sgkit/pull/899) (Tom) Fix max str len for dask

#### Issues

* [sgkit#893](https://github.com/pystatgen/sgkit/issues/893) Optional genomic-scale for popgen stats

#### Attendees

1. Jeff Hammerbacher
2. Tom White
3. Rafal Wojdyla

#### Discussion

* Performance analysis
    * Get familiar with Cubed and Modal
    * Tom has some work to do getting sgkit using Cubed
        * R:
            * Q1: does it make sense to do it right now?
            * Q2: Nvidia blog?
    * [Tom Nicholas](https://github.com/TomNicholas) proposing generalization of chunking API
    * Tom W can give a presentation on Cubed soon
    * map_overlap from Dask could be hard in Cubed
    * GPU stuff and REGENIE probably won’t work in Cubed now
* [sgkit PR#891](https://github.com/pystatgen/sgkit/pull/891)
    * Good to go, but rebase needed
* [sgkit PR#892](https://github.com/pystatgen/sgkit/pull/892)
    * Jerome’s [“not convinced about the general approach of creating new _imputed variables.”](https://github.com/pystatgen/sgkit/pull/892)
* [sgkit PR#895](https://github.com/pystatgen/sgkit/pull/895)
    * Good to go, but rebase needed
* [sgkit PR#899](https://github.com/pystatgen/sgkit/pull/899)
    * A bit confusing what’s going on
    * Will try to add a test
* [sgkit#893](https://github.com/pystatgen/sgkit/issues/893)
    * Tom will put PR together to move forward

**Comment by tomwhite** [comment]
Relevant xarray issue for generalizing chunking backend: https://github.com/pydata/xarray/issues/6807

**Comment by timothymillar** [comment]
Thanks for recording these, it's quite helpful seeing the outline of what's been discussed
